src/Symfony/Component/Security/Http/Firewall/AbstractAuthenticationListener.php

```diff
@@ -176,7 +177,11 @@ protected function requiresAuthentication(Request $request)
     private function onFailure(Request $request, AuthenticationException $failed)
     {
         if (null !== $this->logger) {
-            $this->logger->info('Authentication request failed.', ['exception' => $failed]);
+            if ($failed instanceof AuthenticationServiceException) {
```

Comment:
can't we inline this ?

```php
$errorType = $failed instanceof AuthenticationServiceException ? 'error' : 'info' 
$this->logger->log('$errorType', 'Authentication request failed.', ['exception' => $failed]);
```
I feel it better like this since we don't duplicate the code and the logger allow doing things like that ;).

Reply:
@Simperfit I agree, but with a slight change. It's not type, but level (like in psr logger).